spec/Section 5 -- Validation.md

```diff
@@ -1894,8 +1894,12 @@ AreTypesCompatible(variableType, locationType):
   - Let {nullableVariableType} be the nullable type of {variableType}.
   - Return {AreTypesCompatible(nullableVariableType, locationType)}.
 - Otherwise, if {locationType} is a list type:
-  - If {variableType} is NOT a list type, return {false}.
   - Let {itemLocationType} be the unwrapped item type of {locationType}.
+  - If {variableType} is NOT a list type:
+    - If {itemLocationType} is a non-null type:
+      - Let {nullableItemLocationType} be the unwrapped nullable type of {itemLocationType}.
+      - Return {AreTypesCompatible(variableType, nullableItemLocationType)}.
+    - Otherwise, return {AreTypesCompatible(variableType, itemLocationType)}.
```

Comment:
Note: this is simpler to write and equally as effective (and potentially more understandable):
```suggestion
    - If {itemLocationType} is a non-null type:
      - Let {nonNullVariableType} be the non-null type of {variableType}.
      - Return AreTypesCompatible(nonNullVariableType, itemLocationType).
```
However, all the rest of the logic favors unwrapping types, rather than wrapping types.  So with a single extra line in the logic here, we are only unwrapping types, and never wrapping types.